app.js

```diff
@@ -6,72 +6,68 @@ const Rooms  = require('./rooms.js')
+const userConfig = Input.setName()
+    .then((uname) => {
+        username = uname
+        return Rooms.getRooms()
+    })
+    .then((chatRooms) => {
+        console.log('Available rooms:')
+        if (chatRooms.length) {
+            chatRooms
+                .forEach(room => console.log(`${room.name}: ${room.size} online`))
+        } else {
+            console.log('No current chat rooms, create your own.')
+        }
+        return Input.setRoom()
+    })
+    .catch((err) => {
+        console.log('Error getting rooms: ', err)
+    })
 
 socket.on('connect', (data) => {
-    console.log('Connected to Server')
-
-    Input.setName()
-        .then((uname) => {
-            username = uname
-            return Rooms.getRooms()
-        })
-        .then((chatRooms) => {
-            console.log('Available rooms:')
-            if (chatRooms.length) {
-                chatRooms
-                    .forEach(room => console.log(`${room.name}: ${room.size} online`))
-            } else {
-                console.log('No current chat rooms, create your own.')
-            }
-
-            return Input.setRoom()
-        })
+    userConfig
         .then((room) => {
             socket.emit('create', room)
             console.log(`Joined room: ${room}`)
-            listenForInput()
+
+            Input.setMessage(messageHandler)
+            Input.setPrompt()
         })
         .catch((err) => {
-            console.log('Error getting rooms', err)
+            console.log('Error connecting to room: ', err)
         })
 })
 
-socket.on('invalid', (error) => {
-    console.log('error', error)
-})
-
 socket.on('chat', (data) => {
     const date = new Date(data.date)
     const hour = date.getHours()
     const min = date.getMinutes()
-	console.log('')
-    //TODO: Fix the % sign showing up
+
+    Input.clearLine()
     if (min < 10) {
         process.stdout.write(Colors.cyan +`${hour}:0${min} : ` + Colors.reset + Colors.yellowBgBlackLt +`${data.name}` + Colors.reset +`\n`)
     } else {
         process.stdout.write(Colors.cyan +`${hour}:${min} : ` + Colors.reset + Colors.yellowBgBlackLt +`${data.name}` + Colors.reset +`\n`)
     }
-    console.log(Colors.reset,`- ${data.message}`)
-	Input.resetCursor()
+    process.stdout.write(Colors.reset + `- ${data.message}` + `\n`)
+    Input.setPrompt()
 })
 
-function listenForInput() {
-    Input.setMessage()
-        .then((message) => {
-            if (message === ':q') {
-                process.exit()
-            }
-            socket.emit('chat', {
-                name: username,
-                date: new Date(),
-                message: message
-            })
-			listenForInput() 
-        })
-        .catch((err) => {
-            console.log('Error sending message', err)
-        })
+socket.on('invalid', (err) => {
+    console.log('Error invalid: ', err)
+})
+
+function messageHandler(message) {
+    if (message === ':q') {
+        process.exit()
+    }
```

Comment:
can we throw this into a else just so it looks better(to me) also what happens if the socket.emit fails? What do we do with the error?